storage/gcp/gcp.go

```diff
@@ -158,22 +161,39 @@ func (s *Storage) setTile(ctx context.Context, level, index, logSize uint64, til
 	return s.objStore.setObject(ctx, tPath, data, &gcs.Conditions{DoesNotExist: true})
 }
 
-// getTile returns the tile at the given tile-level and tile-index for the specified log size.
+// getTiles returns the tiles with the given tile-coords for the specified log size.
 //
-// Returns a wrapped os.ErrNotExist if the tile does not exist.
-func (s *Storage) getTile(ctx context.Context, level, index, logSize uint64) (*api.HashTile, error) {
-	objName := layout.TilePath(level, index, logSize)
-	data, _, err := s.objStore.getObject(ctx, objName)
-	if err != nil {
-		if errors.Is(err, gcs.ErrObjectNotExist) {
-			// Return the generic NotExist error so that higher levels can differentiate
-			// between this and other errors.
-			return nil, fmt.Errorf("%v: %w", objName, os.ErrNotExist)
-		}
+// Returns a wrapped os.ErrNotExist if any of the tiles do not exist.
+func (s *Storage) getTiles(ctx context.Context, tileIDs []storage.TileID, logSize uint64) ([]*api.HashTile, error) {
+	r := make([]*api.HashTile, len(tileIDs))
+	errG := errgroup.Group{}
+	for i, id := range tileIDs {
+		i := i
+		id := id
+		errG.Go(func() error {
+			objName := layout.TilePath(id.Level, id.Index, logSize)
+			data, _, err := s.objStore.getObject(ctx, objName)
+			if err != nil {
+				if errors.Is(err, gcs.ErrObjectNotExist) {
+					// Return the generic NotExist error so that higher levels can differentiate
+					// between this and other errors.
+					return fmt.Errorf("%v: %w", objName, os.ErrNotExist)
+				}
+				return err
+			}
+			t := &api.HashTile{}
+			if err := t.UnmarshalText(data); err != nil {
+				return fmt.Errorf("unmarshal(%q): %v", objName, err)
+			}
+			r[i] = t
+			return nil
+		})
+	}
+	if err := errG.Wait(); err != nil {
 		return nil, err
 	}
```

Comment:
Does this cause a problem if the first error is `NotExist` but another one is a real failure?